src/watertap_contrib/seto/core/solar_energy_base.py

```diff
@@ -112,3 +120,205 @@ def calculate_scaling_factors(self):
 
         if callable(self._scaling):
             self._scaling(self)
+
+    def _load_surrogate(self):
+
+        stream = StringIO()
+        oldstdout = sys.stdout
+        sys.stdout = stream
+
+        self.surrogate_blk = SurrogateBlock(concrete=True)
+        self.surrogate = PysmoSurrogate.load_from_file(self.surrogate_file)
+        self.surrogate_blk.build_model(
+            self.surrogate,
+            input_vars=self.surrogate_inputs,
+            output_vars=self.surrogate_outputs,
+        )
+        sys.stdout = oldstdout
+
+    def _create_rbf_surrogate(
+        self,
+        input_bounds,
+        data_training=None,
+        n_samples=100,
+        training_fraction=0.8,
+        dataset_filename=None,
+        output_filename=None,
+        input_labels=None,
+        output_labels=None,
+        build_model=False,
+    ):
+        if not hasattr(self, "input_labels"):
+            self.input_labels = input_labels
+        if not hasattr(self, "output_labels"):
+            self.output_labels = output_labels
+
+        if data_training is None:
+            self._get_surrogate_data(
+                dataset_filename=dataset_filename,
+                n_samples=n_samples,
+                training_fraction=training_fraction,
+            )
+        else:
+            self.data_training = data_training
+
+        # Capture long output
+        stream = StringIO()
+        oldstdout = sys.stdout
+        sys.stdout = stream
+
+        # Create PySMO trainer object
+        self.trainer = PysmoRBFTrainer(
+            input_labels=self.input_labels,
+            output_labels=self.output_labels,
+            training_dataframe=self.data_training,
+        )
+
+        # Set PySMO options
+        self.trainer.config.basis_function = "gaussian"  # default = gaussian
+        self.trainer.config.solution_method = "algebraic"  # default = algebraic
+        self.trainer.config.regularization = True  # default = True
```

Comment:
I wonder if these should be config options or at least more accessible for the user to change as desired.